src/framework/ui/datepicker/baseDatepicker.component.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import {
+  ImageStyle,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  ImageProps,
+  StyleSheet,
+  GestureResponderEvent,
+} from 'react-native';
+import {
+  Text,
+  TextElement,
+} from '../text/text.component';
+import { Popover } from '../popover/popover.component';
+import { CalendarElement } from '../calendar/calendar.component';
+import { RangeCalendarElement } from '../calendar/rangeCalendar.component';
+import {
+  Interaction,
+  StyledComponentProps,
+  StyleType,
+} from '@kitten/theme';
+import { Dimensions } from 'react-native';
+
+interface State {
+  visible: boolean;
+}
+
+export interface ComponentProps {
+  icon?: (style: ImageStyle) => React.ReactElement<ImageProps>;
+}
+
+export type DatepickerProps<D> = ComponentProps & StyledComponentProps & TouchableOpacityProps;
+
+export abstract class BaseDatepickerComponent<D, P> extends React.Component<DatepickerProps<D> & P, State> {
+
+  public state: State = {
+    visible: false,
+  };
+
+  public abstract getComponentTitle(): string;
+
+  public abstract renderCalendar(): CalendarElement<D> | RangeCalendarElement<D>;
+
+  private getComponentStyles = (style: StyleType): StyleType => {
+    const {
+      popoverMarginHorizontal,
+      textFontSize,
+      textLineHeight,
+      textFontWeight,
+      textColor,
+      iconWidth,
+      iconHeight,
+      iconTintColor,
+      ...containerStyles
+    } = style;
+
+    return {
+      container: containerStyles,
+      icon: {
+        width: iconWidth,
+        height: iconHeight,
+        tintColor: iconTintColor,
+      },
+      text: {
+        fontSize: textFontSize,
+        lineHeight: textLineHeight,
+        fontWeight: textFontWeight,
+        color: textColor,
+      },
+      popover: {
+        indent: popoverMarginHorizontal,
+      },
+    };
+  };
+
+  private onPressIn = (event: GestureResponderEvent) => {
+    this.props.dispatch([Interaction.ACTIVE]);
+
+    if (this.props.onPressIn) {
+      this.props.onPressIn(event);
+    }
+  };
+
+  private onPressOut = (event: GestureResponderEvent) => {
+    this.props.dispatch([]);
+
+    if (this.props.onPressOut) {
+      this.props.onPressOut(event);
+    }
+  };
+
+  private toggleVisible = (): void => {
+    const visible: boolean = !this.state.visible;
+
+    this.setState({ visible }, this.dispatchActive);
+  };
+
+  private dispatchActive = (): void => {
+    const { visible } = this.state;
+    if (visible) {
+      this.props.dispatch([Interaction.ACTIVE]);
+    } else {
+      this.props.dispatch([]);
+    }
+  };
+
+  public formatDateToString(date: D): string {
+    const selectedDate: Date = new Date(Date.parse(date.toString()));
+    const day: number = selectedDate.getDate();
+    const month: number = selectedDate.getMonth() + 1;
+    const year: number = selectedDate.getFullYear();
+
+    return `${day}/${month}/${year}`;
```

Comment:
use date service to format dates and support custom dates like moment, like we do it in Calendar component